app/code/FrontCommerce/ChronorelaisHeadless/Model/RelayPointManagement.php

```diff
@@ -0,0 +1,113 @@
+{
+    protected const METHOD_CODE = 'chronorelais';
+
+    protected RelayPoint $relayPoint;
+    protected RelayPointsResponseInterface $relayPointsResponse;
+    protected Session $session;
+    protected CartRepositoryInterface $cartRepository;
+
+    public function __construct(
+        RelayPoint                   $relayPoint,
+        RelayPointsResponseInterface $relayPointsResponse,
+        Session                      $session,
+        CartRepositoryInterface      $cartRepository
+    )
+    {
+        $this->relayPoint = $relayPoint;
+        $this->relayPointsResponse = $relayPointsResponse;
+        $this->cartRepository = $cartRepository;
+        $this->session = $session;
+    }
+
+    /**
+     * @throws Exception
+     */
+    public function getRelayPoints(string $postCode, array $shippingAddress): array
+    {
+        $this->checkParams($postCode, $shippingAddress);
+
+        return $this->formatRelayResponseData(
+            $this->relayPoint->getRelayPoints(self::METHOD_CODE, $postCode, $shippingAddress)
+        );
+    }
+
+    protected function formatRelayResponseData(array $relayPoints): array
+    {
+        if (!$relayPoints) {
+            return [];
+        }
+
+        $result = [];
+        $relayPoints = json_decode(json_encode($relayPoints));
+
+        foreach ($relayPoints as $relayPoint) {
+            $relayPoint = json_decode(json_encode($relayPoint));
+
+            $result[] = clone $this->relayPointsResponse
+                ->setStreet1($relayPoint->adresse1)
+                ->setStreet2($relayPoint->adresse2)
+                ->setStreet3($relayPoint->adresse3)
+                ->setLatitude($relayPoint->latitude)
+                ->setLongitude($relayPoint->longitude)
+                ->setZipCode($relayPoint->codePostal)
+                ->setRelayId($relayPoint->identifiantChronopostPointA2PAS)
+                ->setName($relayPoint->nomEnseigne)
+                ->setCity($relayPoint->localite)
+                ->setName($relayPoint->nomEnseigne)
+                ->setOpeningTime([
+                    'Monday' => $relayPoint->horairesOuvertureLundi,
+                    'Tuesday' => $relayPoint->horairesOuvertureMardi,
+                    'Wednesday' => $relayPoint->horairesOuvertureMercredi,
+                    'Thursday' => $relayPoint->horairesOuvertureJeudi,
+                    'Friday' => $relayPoint->horairesOuvertureVendredi,
+                    'Saturday' => $relayPoint->horairesOuvertureSamedi,
+                    'Sunday' => $relayPoint->horairesOuvertureDimanche,
+                ]);
+        }
+
+        return $result;
+    }
+
+    /**
+     * @throws Exception
+     */
+    protected function checkParams(string $postCode, array $shippingAddress)
+    {
+        if (!$postCode) {
+            throw new Exception(__('Missing post_code'));
+        }
+
+        if (!isset($shippingAddress['country_id'])) {
```

Comment:
this does not follow what is written in the ticket.

The API is supposed to respond to `GET /rest/default/V1/chronorelais/relais?country_id=<COUNTRYCODE>&postcode=<ZIPCODE>` with that it expects `GET /rest/default/V1/chronorelais/relais?shippingAddress[country_id]=<COUNTRYCODE>&postcode=<ZIPCODE>`